ui/components/ui/show-hide-toggle/show-hide-toggle.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import IconEye from '../icon/icon-eye';
+
+const ShowHideToggle = ({
+  id,
+  checked,
+  onChange,
+  ariaLabel,
+  className,
+  dataTestId,
+  disabled,
+}) => {
+  return (
+    <div className={classnames('show-hide-toggle', className)}>
+      <input
+        className="show-hide-toggle__input"
+        id={id}
+        type="checkbox"
+        checked={checked}
+        onChange={onChange}
+        data-testid={dataTestId}
+        disabled={disabled}
+      />
+      <label htmlFor={id} className="show-hide-toggle__label">
+        <IconEye ariaLabel={ariaLabel} className="show-hide-toggle__icon" />
```

Comment:
Looks like this doesn't display any visual indication when it is focused. Or maybe it's the checkbox that is getting focused, and it's not shown because it's off screen. Either way, we have to ensure there is a visual representation of focus for the sake of helping keyboard users.

Curious why you chose to go this route of separating the checkbox component from the label, hiding the checkbox, and using the label instead. This seems like a lot more work than using CSS to display the checkbox as an eye. It's not clear this is any better for accessibility, readability, or even compatibility.

Reply:
I should double check this is an acceptable standard for a custom checkbox but I believed it was. I do see a visual representation of focus in my storybook do you see something different. I did change the component api in another update


https://user-images.githubusercontent.com/8112138/158444898-ba3d5072-4b14-4f85-b713-bf11341ac4b8.mov

Reply:
No haven't browser tested and haven't written unit tests yet. I'll do that now

Reply:
This was fixed by replacing the webkit-specific highlight color referenced in the styles with `Highlight`, which should work across browsers.

Also George explained to me the evils of icon fonts, so I'm now convinced that this is a much better approach than what I had put together before this :sweat_smile: